app/queries/proxy_rule_query.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ProxyRuleQuery
+  def initialize(owner_type:, owner_id:, sort: nil, direction: nil)
+    @direction  = direction
+    @owner_type = owner_type
+    @owner_id   = owner_id
+    @sort       = sort
+  end
+
+  def search_for(query, scope = ProxyRule.all)
+    scope = scope.order_by(@sort, @direction).includes(:metric)
+    return scope if query.blank?
+    options = {
+      ids_only: true, star: true, per_page: 1_000_000, ignore_scopes: true,
```

Comment:
1 million per page? Are you sure?

Reply:
yeah.. don't make sense haha.. will add the same maximum as we paginate.